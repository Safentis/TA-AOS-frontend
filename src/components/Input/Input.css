.input {
  border: none;
  border-radius: 0.3rem;
  box-shadow: 0.15rem 0.15rem 0.15rem var(--main-dk-grey-color);
  color: var(--main-dark-blue-color);
  outline: none;
  padding: 1rem 1.5rem;
}

.input,
.input::placeholder {
  font-size: 1.4rem;
  font-weight: 400;
}
