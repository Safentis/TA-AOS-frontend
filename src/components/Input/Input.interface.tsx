import React from 'react';

export interface Props {
  className?: string;
  value: string;
  id: string;
  handleChange: (event: React.ChangeEvent) => void;
  placeholder: string;
}
