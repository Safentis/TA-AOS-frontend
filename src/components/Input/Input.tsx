import React, { FC } from 'react';

import './Input.css';
import { Props } from './Input.interface';

const Input: FC<Props> = ({
  className = '',
  value,
  id,
  placeholder,
  handleChange,
}): React.ReactElement => {
  return (
    <input
      className={'input ' + className}
      placeholder={placeholder}
      onChange={handleChange}
      value={value}
      type="text"
      id={id}
    />
  );
};

export default Input;
