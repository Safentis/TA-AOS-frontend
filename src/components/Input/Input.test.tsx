import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Input from './Input';

import { Props } from './Input.interface';

const setUp = (props: Props) => {
  return shallow(<Input {...props} />);
};

describe('<Input />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
    value: 'string',
    id: 'some-id',
    placeholder: 'some-text',
    handleChange: () => {},
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 input element', () => {
      expect(component.find('input')).toBeTruthy();
    });

    it('has style "input" and "custom-style"', () => {
      expect(component.find('input').hasClass('input')).toBeTruthy();
      expect(
        component.find('input').hasClass('custom-style'),
      ).toBeTruthy();
    });

    it(`has value ${props.value}`, () => {
      expect(
        component.find('input').prop('value') === props.value,
      ).toBeTruthy();
    });
  });
});

describe('<Input /> default props', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    value: 'string',
    id: 'some-id',
    placeholder: 'some-text',
    handleChange: () => {},
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has only one style "input"', () => {
      expect(component.find('input').hasClass('input')).toBeTruthy();
    });
  });
});
