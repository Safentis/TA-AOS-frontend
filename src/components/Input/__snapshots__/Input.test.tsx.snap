// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component does match to the snapshot 1`] = `
<input
  className="input custom-style"
  id="some-id"
  onChange={[Function]}
  placeholder="some-text"
  type="text"
  value="string"
/>
`;

exports[`<Input /> default props component does match to the snapshot 1`] = `
<input
  className="input "
  id="some-id"
  onChange={[Function]}
  placeholder="some-text"
  type="text"
  value="string"
/>
`;
